src/main/java/me/karakelley/http/responses/BadRequest.java

```diff
@@ -0,0 +1,10 @@
+package me.karakelley.http.responses;
+
+import me.karakelley.http.Response;
+
+public class BadRequest extends Response {
```

Comment:
You'll find a lot of wariness around inheritance at 8th Light, for good reason. I like what you're going for here, but I think there are some alternatives to the inheritance approach. Tangentially related: check out [enums](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html)—might be a nice way to model your response codes.

Reply:
I was hesitant to use inheritance but the reason I chose it was because it allowed me to dispatch a particular response type without having to worry about setting a particular status, etc. For instance, in `ConnectionHandler`, I can hand off a response object to `ResponseWriter` and let it do its thing.

```java 
 private void handleRequest(LineReader reader, OutputStream out, Handler handler, RequestParser requestParser, Socket clientSocket) {
    Response response;
    try {
      Request request = requestParser.parse(reader).buildRequest(clientSocket.getLocalPort());
      response = handler.respond(request);
      new ResponseWriter(out, response).deliver();
    } catch (InvalidRequestException e) {
      new ResponseWriter(out, new BadRequest()).deliver();
    }
  }
```

 I wanted to reach for enums but had a hard time figuring out how to use enums and keep my validations at this high level without worrying about implementation details.